### PR TITLE
Add YAML file extensions to pre-commit hooks config

### PR DESCRIPTION
**Pull request by cas--**
- Prettier support YAML but pre-commit hook was not checking those
  files so added file extensions to hooks config.
- Used double-quoted flow scalar style to make the list of file
  extensions easier to read. Backslash for regex needed escaping and
  backslash each line to prevent YAML including spaces.
- Added alternative markdown file extensions.
- Formatted with prettier.

### CODE REVIEW

**Comment by azz** [comment]
Missing `.vue`?

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+    |scss\
+    |ts|tsx\
+    |graphql|gql\
+    |json\
+    |js\
+    |jsx\
+    |md|markdown|mdown|mkdn\
+    |mdx\
+    |vue\
+    |yaml|yml\
```
[the remark this answers is not in the document]
Comment (cas--) [reply .pre-commit-hooks.yaml]: sorry what do you mean by that?

[the remark this answers is not in the document]
Comment (lipis) [reply .pre-commit-hooks.yaml]: sort alphabetically.. a-z

**Comment by cas--** [comment]
@lipis Fixed comments and merged to single commit

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+- id: prettier
+  name: prettier
+  entry: prettier --write
+  language: node
+  files: "\\.(\
+    css|less|scss\
+    |graphql|gql\
+    |js\
+    |json\
+    |jsx\
```
Comment (lipis) [inline .pre-commit-hooks.yaml]: one last thing.. js and jsx could be grouped as well :) no need to merge in one commit.. we are squashing anyway

Comment (cas--) [reply .pre-commit-hooks.yaml]: Done :)

**Comment by lipis** [comment]
waiting for the robots to give the green light now

**Comment by ikatyang** [comment]
Thanks!

**Pull request merged**